tests/e2e/multigpu/test_locking.py

```
@@ -0,0 +1,196 @@
+
+        def mock_exists():
+            nonlocal exists_call_count
+            exists_call_count += 1
+
+            if exists_call_count == 1:
+                # First check: ready flag exists (not first process)
+                return True
+            if exists_call_count <= 3:
+                # While loop checks: flag doesn't exist yet
+                return False
+            return True
```

Ruff (0.11.9)

Return the negated condition directly

(SIM103)